docs/tutorials/particle-filtering.md

```diff
@@ -1,3 +1,451 @@
+
+```cpp
+double current_position{parameters.initial_position};
+for (std::size_t n = 0; n < parameters.number_of_cycles; ++n) {
+  current_position += parameters.velocity * parameters.dt;
+
+  if (current_position > static_cast<double>(parameters.map_size)) {
+    break;
+  }
+
+  // Rest of the code
+  // ...
+}
+```
+
+### 1.5 Prediction
+
+<!-- TODO(alon): What is it a motion model explanation should be in another doc like 'extending-beluga' -->
+For the prediction step, we will be using a **Sample Motion Model Odometry** (_Probabilistic Robotics [^ProbRob], page 111_). Technically, odometry are sensors measurements, not controls (they are only available retrospectively, after the robot has moved), but this poses no problem for filter algorithms, and it will be treated as a control signal.
+
+At time $t$, the robot's position is a random variable $x(t)$. This model considers that within the time interval $(t-1, t]$, the registered movement from odometry is a reliable estimator of the true state movement, despite any potential drift and slippage that the robot may encounter during this interval.
+
+For the **Sample Motion Model Odometry** we will use the following definitions:
+
+* `particle_position` represent a single particle from the range $x_{t-1}$ before applying the control update step.
+* `distance = parameters.velocity * parameters.dt` is the equivalent of the odometry translation $u_t$.
+* The uncertainty associated with the robot reaching its intended position after a control command is issued is modeled using a Gaussian distribution
+  `std::normal_distribution<double> distribution`.
+* The spread of the Gaussian distribution is controlled by a standard deviation, which is a constant `parameters.translation_sd` that can be specified when initializing the model.
+* The motion model returns a single particle's state corresponding to the new range $x'_t$.
+
+<!-- TODO(alon): The explanation of beluga::actions::propagate should be in another doc like 'extending-beluga' -->
+
+```cpp
+auto motion_model = [&](double particle_position, auto& random_engine) {
+  const double distance = parameters.velocity * parameters.dt;
+  std::normal_distribution<double> distribution(distance, parameters.translation_sd);
+  return particle_position + distribution(random_engine);
+};
+
+
+```
+
+To apply this motion model to all the `particles` range we use `beluga::actions::propagate` function.
+
+```cpp
+particles |= beluga::actions::propagate(std::execution::seq, motion_model);
+```
+
+:::{note}
+The range $x'_t$ represent the posterior $bel'(x'_t)$ before incorporating the measurement $z_t$. This posterior or particle set distribution
+does not match the updated belief $bel(x_t)$ explain in [key concepts](../concepts/key-concepts.md).
+:::
+
+:::{hint}
+The effect of the motion model is to increase the space occupied by the particles.
```

Comment:
@Alondruck hmm, it is true that the motion model will not reduce the dispersion of the prior belief, but talking about increase and space seems like too much. I'd remove the hint.